### PR TITLE
ocamlPackages.minttea: init at 0.0.1

### PR DESCRIPTION
**Pull request by Sixstring982**
## Description of changes

Adding ocaml-ng.ocamlPackages_5_1.minttea, which is built from v0.0.1: https://github.com/leostera/minttea/releases/tag/0.0.1

`minttea` is a TUI builder for OCaml, written by @leostera:
https://github.com/leostera/minttea

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16977,6 +16977,12 @@
     githubId = 8017899;
     name = "Sivaram Balakrishnan";
   };
+  sixstring982 = {
+    email = "sixstring982@gmail.com";
+    github = "sixstring982";
+    githubId = 1328977;
+    name = "Trent Small";
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:16984]: please move this into it's own commit called `maintainers: add sixstring982

File: pkgs/development/ocaml-modules/minttea/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/leostera/minttea/releases/download/${version}/minttea-${version}.tbz";
+    hash = "sha256-Ca0wa6cYs+cuQr2igB7ZhtQ0Dtk=";
+  };
+
+  propagatedBuildInputs = [
+    mdx
+    riot
+    tty
```
Comment (Janik-Haag) [inline pkgs/development/ocaml-modules/minttea/default.nix:23]:
```suggestion
    tty
```
there is no tty package, did you mean to add coreutils?

**Review by Janik-Haag — COMMENTED** [review]
Thank you for your contribution, but the package isn't building, did you try to build it locally?
The commit should be named `ocamlPackages.minttea: init at 0.0.1`

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Sixstring982) [reply maintainers/maintainer-list.nix:16984]: Sounds good -- Done!

File: pkgs/development/ocaml-modules/minttea/default.nix (hunk 2 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/minttea/default.nix:23]: Perhaps -- let me figure out how to build locally, that will help me figure out what's going on here.

Comment (Sixstring982) [reply pkgs/development/ocaml-modules/minttea/default.nix:23]: Looks like `tty` isn't published in `nixpkgs` yet either -- I can open up another PR to do that, as well!

**Comment by Sixstring982** [comment]
Looks like I need to do a few things before this PR is ready for review:

- [ ] Add myself as a maintainer (I'll go ahead and do that in a separate PR)
- [ ]  Add `tty@0.0.1` as an OCaml package

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Sixstring982) [reply maintainers/maintainer-list.nix:16984]: On second thought, I'll do this in a separate PR.

File: pkgs/development/ocaml-modules/minttea/default.nix (hunk 2 above)
Comment (Janik-Haag) [reply pkgs/development/ocaml-modules/minttea/default.nix:23]: > Looks like tty isn't published in nixpkgs yet either -- I can open up another PR to do that, as well!

It is it's in the gnucoreutils package.

> Perhaps -- let me figure out how to build locally, that will help me figure out what's going on here.

You can use `nix-build -A ocamlPackages.minttea` it will give you a `./result/` directory if the run is successful

Comment (Sixstring982) [reply pkgs/development/ocaml-modules/minttea/default.nix:23]: Got it -- thanks! Yes, looks like this builds successfully now (I think!)

**Review by Janik-Haag — APPROVED** [review]

File: pkgs/development/ocaml-modules/minttea/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  minimalOCamlVersion = "5.1";
+
+  src = fetchurl {
+    url = "https://github.com/leostera/minttea/releases/download/${version}/minttea-${version}.tbz";
+    hash = "sha256-+4nVeYKx2A2i2nll/PbStcEa+Dvxd0T7e/KsdJqY4bI=";
+  };
+
+  propagatedBuildInputs = [
+    mdx
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/minttea/default.nix:21]:
```suggestion
```

The `mdx` library should not be propagated.

Are you sure it is even needed at build time?

Comment (Sixstring982) [reply pkgs/development/ocaml-modules/minttea/default.nix:21]: Good point -- I think this is only required for testing. Let me remove this.

Comment (Sixstring982) [reply pkgs/development/ocaml-modules/minttea/default.nix]: OK, I removed `mdx`. Looks like I can build without it!
